netcdf4/src/main/java/ucar/nc2/jni/netcdf/Nc4Iosp.java

```diff
@@ -2339,8 +2350,10 @@ private void createGroup(Group4 g4) throws IOException {
 
     // dimensions
     for (Dimension dim : g4.g.getDimensions()) {
-      int dimid = addDimension(g4.grpid, dim.getShortName(), dim.getLength());
+      int dimLength = dim.isUnlimited() ? Nc4prototypes.NC_UNLIMITED : dim.getLength();
+      int dimid = addDimension(g4.grpid, dim.getShortName(), dimLength);
```

Comment:
And yeah, you'll get an error which is COMPLETELY unhelpful in determining the cause. That was fun to unwind.